sys/kern/src/arch/riscv32.rs

```diff
@@ -435,11 +435,17 @@ fn timer_handler() {
                 set_current_task(next);
             }
 
-            // Reset mtime back to 0.  In theory we could save an instruction on
-            // RV32 here and only write the low-order bits, assuming that it has
-            // been less than 12 seconds or so since our last interrupt(!), but
-            // let's avoid any possibility of a nasty surprise.
-            core::ptr::write_volatile(crate::startup::MTIME as *mut u64, 0);
+            //
+            // Increase mtimecmp for the next interrupt
+            // This will also clear the pending timer interrupt.
+            //
+            let mtimecmp =
+                core::ptr::read_volatile(crate::startup::MTIME as *mut u64);
```

Comment:
Should be `mtimecmp`.

Reply:
Umm.. Is it because you want to ensure that the interrupt happens exactly `CLOCK_FREQ_KHZ` after the last timer interrupt i.e. last value of `MTIMECMP`?

Reply:
Yea, the way it is, the intervals wont be exact.  Was just a typo on my part.

Reply:
Oh okay. In practice, I think it shouldn't really matter much if the intervals are not exact. If you don't care about the exact intervals, then you can simplify the handling of `set_timer` i.e. always read mtime, add CLOCK_FREQ_KHZ and write it to mtimecmp. Else, you will need a special case to check if mtimecmp is 0, then read mtime else use mtimecmp.

Reply:
Hmm, I guess if it is okay to lose a couple of ticks, it just seems strange.  I was just going to set both `MTIME` and `MTIMECMP` to zero at startup.